src/test/java/com/flagsmith/FlagsmithCachedApiWrapperTest.java

```diff
@@ -11,554 +12,197 @@
 import static org.testng.Assert.assertNull;
 import static org.testng.Assert.assertThrows;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.config.FlagsmithConfig;
+import com.flagsmith.config.Retry;
+import com.flagsmith.flagengine.features.FeatureStateModel;
+import com.flagsmith.flagengine.identities.traits.TraitModel;
+import com.flagsmith.interfaces.FlagsmithCache;
+import com.flagsmith.models.Flags;
+import com.flagsmith.threads.RequestProcessor;
 import com.github.benmanes.caffeine.cache.Cache;
 import com.github.benmanes.caffeine.cache.Caffeine;
 import java.util.ArrayList;
+import java.util.List;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.mock.MockInterceptor;
+import org.mockito.ArgumentCaptor;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 @Test(groups = "unit")
 public class FlagsmithCachedApiWrapperTest {
```

Comment:
I kept this class as separate so that we have some separation of cached related tests for flagsmithApiWrapper. Also, this way we can have some simple and readable implementation of the init method without much complexity.

Reply:
Ok, understood. Could we perhaps update the name of the test class though to avoid confusion? Maybe something like `FlagsmithApiWrapperCachingTest` ?